src/node_api.cc

```diff
@@ -775,10 +775,16 @@ napi_status napi_define_class(napi_env env,
       t->SetClassName(property_name);
 
       tpl->PrototypeTemplate()->Set(property_name, t, attributes);
-    } else if (p->getter || p->setter) {
+    } else if (p->getter != nullptr || p->setter != nullptr) {
       v8::Local<v8::Object> cbdata = v8impl::CreateAccessorCallbackData(
         env, p->getter, p->setter, p->data);
 
+      // The napi_writable attribute is ignored for accessor descriptors,
+      // but V8 requires the ReadOnly attribute to match existence of a setter.
+      attributes = static_cast<v8::PropertyAttribute>(p->setter != nullptr
+        ? attributes & ~v8::PropertyAttribute::ReadOnly
+        : attributes | v8::PropertyAttribute::ReadOnly);
```

Comment:
Yeah I sort of wanted to do that, but it didn't seem right to put it in `v8impl::V8PropertyAttributesFromAttributes()` because the logic depends on the kind of property descriptor. I'll consider further what to do with this.